cmake/utils.cmake

```diff
@@ -401,15 +321,17 @@ function(separate_argument_list listName)
 endfunction()
 
 # python extension module suffix
-function(set_python_module_suffix target)
+function(set_python_module_property target)
```

Comment:
This new function name is a little mysterious and looks a bit weird in CMakeLists.txt files below. What property? :P

It's a bit verbose, but maybe `set_python_module_suffix_property()`? Or do we expect that other properties might need to be added here? If that's the case maybe `set_python_module_properties()`?

And/or possibly prefix this with `mayaUsd_`? I'd be mildly concerned that this could be mistaken for a built-in CMake function.

Reply:
`set_python_module_property ` is simply a tiny wrapper around cmake `set_target_properties` function that sets both `prefix ` and `suffix ` names for python modules on different platforms.

```
function(set_python_module_property target)
    if(IS_WINDOWS)
        set_target_properties(${target}
            PROPERTIES
                PREFIX ""
                SUFFIX ".pyd"
        )
    elseif(IS_LINUX OR IS_MACOSX)
        set_target_properties(${target}
            PROPERTIES
                PREFIX ""
                SUFFIX ".so"
        )
    endif()
endfunction()
```
Every cmake target (shared library, execturable,...) has some properties that can be modified through `set_target_properties` function. 

Here is the list of all the properties on targets:

https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#properties-on-targets

Besides Targets, (Tests, Directories, Source Files, Cache Entries) can also have their own properties which are all listed here:

https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#id1

I totally agree with the name I chose being off. :)  What do you think about 

`mayaUsd_set_python_module_properties`

Reply:
Yup, I think it was mostly the name that tripped me up. `mayaUsd_set_python_module_properties` sounds great to me.